File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int[] numberList;
    int targerNumber;
    
    public int solution(int[] numbers, int target) {
        numberList = numbers;
        targerNumber = target;
        
        return dfs(0, 0);
    }
    
    public int dfs(int cnt, int result) {
        
        if (cnt >= numberList.length) {
            if (targerNumber == result) {
                return 1;
            }
            return 0;
        }
        
        int answer = 0;
        
        answer += dfs(cnt + 1, result - numberList[cnt]);
        answer += dfs(cnt + 1, result + numberList[cnt]);
            
        return answer;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
from collections import deque

def solution(numbers, target):
    answer = 0
    n = len(numbers)
    que = deque()
    # (idx, 현재값)
    que.append((0, 0))
    
    while que:
        idx, value = que.popleft()
        # idx가 끝까지 가면 종료
        if idx == n:
            if value == target:
                answer+=1
            continue
        
        que.append((idx+1, value - numbers[idx]))
        que.append((idx+1, value + numbers[idx]))
        
    
    return answer